pages/referrals/status/[id].js

```diff
@@ -1,21 +1,72 @@
-import { useState, useEffect } from 'react';
 import { requestByLinkId } from 'lib/api';
 import HttpStatusError from 'lib/api/domain/HttpStatusError';
 import { REFERRAL_STATUSES } from 'lib/utils/constants';
+import useReferral from 'lib/api/utils/useReferral';
+import { IsoDateTime } from 'lib/domain/isodate';
+import { useState } from 'react';
+import StatusForm from 'components/Feature/StatusForm';
+import Head from 'next/head';
+import { convertIsoDateToDateTimeString } from 'lib/utils/date';
 
 const StatusHistory = ({ referral }) => {
+  const { updateReferral } = useReferral(null, {});
+  const [initialReferral, setInitialReferral] = useState(referral);
+  const [submitted, setSubmitted] = useState(false);
+
   const maxDate = new Date(Math.max(...referral.statusHistory.map(e => new Date(e.date))));
-  const recentStatus = referral.statusHistory.filter(status => new Date(status.date) != maxDate)[0];
+  const recentStatus = referral.statusHistory.filter(
+    status => new Date(status.date).getTime() === maxDate.getTime()
+  )[0];
+
+  const onSubmitForm = async e => {
+    e.preventDefault();
+
+    const newStatus = e.target['referral-status'].value;
+    const newStatusHistory = initialReferral.statusHistory.concat([
+      {
+        status: newStatus,
+        date: IsoDateTime.now(),
+        comment: e.target['referral-rejection-reason'].value
+      }
+    ]);
+    const newReferral = {
+      id: initialReferral.id,
+      statusHistory: newStatusHistory
+    };
+
+    await updateReferral(newReferral);
+    setSubmitted(true);
+  };
 
   return (
-    <div>
-      {recentStatus.status == REFERRAL_STATUSES.Approved && (
-        <div>The referral was approved on {recentStatus.date}</div>
-      )}
-      {recentStatus.status != REFERRAL_STATUSES.Approved && (
-        <div>The status of the referral is {recentStatus.status}</div>
+    <>
+      <Head>
+        <title>Status page</title>
+      </Head>
+      {submitted ? (
+        <div className="govuk-grid-row">
+          <div className="govuk-grid-column-two-thirds">
+            <div
+              className="govuk-panel govuk-panel--confirmation"
+              data-testid="status-confirmation-panel">
+              <h1 className="govuk-panel__title">Thank you</h1>
+              <div className="govuk-panel__body">Your decision on this referral has been sent</div>
+            </div>
+            <h2 className="govuk-heading-m">What happens next</h2>
+            <p>We’ve let the referrer know your response.</p>
+          </div>
+        </div>
+      ) : recentStatus.status == REFERRAL_STATUSES.Accepted ||
+        recentStatus.status == REFERRAL_STATUSES.Rejected ? (
+        <div data-testid="status-paragraph">
+          This referral was {recentStatus.status} on{' '}
+          {convertIsoDateToDateTimeString(new Date(recentStatus.date))}
+          {recentStatus.comment && ` with comment "${recentStatus.comment}".`}
```

Comment:
Maybe we should add a colon for the wording to be: `with comment: "some comment"`?

Reply:
good shout!